components/recurring-contributions/UpdatePaymentMethodPopUp.js

```diff
@@ -0,0 +1,287 @@
+import React, { Fragment, useEffect, useState } from 'react';
+import PropTypes from 'prop-types';
+import { useMutation, useQuery } from '@apollo/react-hooks';
+import { PlusCircle } from '@styled-icons/boxicons-regular/PlusCircle';
+import { MoneyCheck } from '@styled-icons/fa-solid/MoneyCheck';
+import themeGet from '@styled-system/theme-get';
+import gql from 'graphql-tag';
+import { first, get, uniqBy } from 'lodash';
+import { withRouter } from 'next/router';
+import { defineMessages, FormattedDate, FormattedMessage, useIntl } from 'react-intl';
+import styled from 'styled-components';
+
+import { formatCurrency } from '../../lib/currency-utils';
+import { getErrorFromGraphqlException } from '../../lib/errors';
+import { API_V2_CONTEXT, gqlV2 } from '../../lib/graphql/helpers';
+import { getPaymentMethodName, paymentMethodExpiration } from '../../lib/payment_method_label';
+
+import { Box, Flex } from '../Grid';
+import CreditCard from '../icons/CreditCard';
+import GiftCard from '../icons/GiftCard';
+import LoadingPlaceholder from '../LoadingPlaceholder';
+import StyledButton from '../StyledButton';
+import StyledHr from '../StyledHr';
+import StyledRadioList from '../StyledRadioList';
+import { P } from '../Text';
+
+const PaymentMethodBox = styled(Flex)`
+  border-top: 1px solid ${themeGet('colors.black.300')};
+`;
+
+const messages = defineMessages({
+  cancel: {
+    id: 'actions.cancel',
+    defaultMessage: 'Cancel',
+  },
+  update: {
+    id: 'subscription.updateAmount.update.btn',
+    defaultMessage: 'Update',
+  },
+  updatePaymentMethod: {
+    id: 'subscription.menu.editPaymentMethod',
+    defaultMessage: 'Update payment method',
+  },
+});
+
+const getPaymentMethodsQuery = gql`
+  query Collective($collectiveSlug: String) {
```

Comment:
I'm working on a RFC to for GraphQL operation names. `query Collective` here is too generic. It should ideally be a name that is unique in the codebase.

Reply:
`UpdatePaymentMethodPopUpQuery` for example.